tests/matrix_multiply_tests/matrix_multiply.futil

```diff
@@ -0,0 +1,232 @@
+import "primitives/core.futil";
+import "primitives/memories/comb.futil";
+import "primitives/binary_operators.futil";
+import "primitives/unsynthesizable.futil";
+
+// Modified from pass-in-register.futil
```

Comment:
We talked a bit about `pass-in-register.futil`, but I just wanted to reiterate that I appreciate that you're learning from existing Calyx code and you've credited the original file! :slightly_smiling_face: